MonoGame.Framework/Vector3.cs

```diff
@@ -351,12 +353,9 @@ public static Vector3 Cross(Vector3 vector1, Vector3 vector2)
         /// <param name="result">The cross product of two vectors as an output parameter.</param>
         public static void Cross(ref Vector3 vector1, ref Vector3 vector2, out Vector3 result)
         {
-            var x = vector1.Y * vector2.Z - vector2.Y * vector1.Z;
-            var y = -(vector1.X * vector2.Z - vector2.X * vector1.Z);
-            var z = vector1.X * vector2.Y - vector2.X * vector1.Y;
-            result.X = x;
-            result.Y = y;
-            result.Z = z;
+            result.X = vector1.Y * vector2.Z - vector2.Y * vector1.Z;
+            result.Y = -(vector1.X * vector2.Z - vector2.X * vector1.Z);
+            result.Z = vector1.X * vector2.Y - vector2.X * vector1.Y;
```

Comment:
Using the local variables was for a reason.  The same variable may be used for vector1 or vector2 and for the result.  Your changes here will result in an incorrect cross product because later calculations will be affected by earlier calculations.  Calculate the cross product into a separate vector (our three local variables) and then assign to the result.